compiler/src/compile.rs

```diff
@@ -51,6 +52,15 @@ impl CallType {
     }
 }
 
+// See https://docs.python.org/3/library/constants.html#built-in-constants
+static BUILTIN_CONSTANTS: phf::Set<&'static str> = phf::phf_set! {
```

Comment:
A constant set seems like its too much here. There's only a couple of constants (Python has 4, `None, True, False, __debug__`, and usually skips checking all of them), a slice would probably do the trick without needing an additional crate.